### PR TITLE
feat: use rl chat cmds plugin as additional kc source

### PR DESCRIPTION
**Pull request by iProdigy**
Supersedes #335
Refactors kill count tracking to a separate class to work towards #371

### CODE REVIEW

File: src/main/java/dinkplugin/util/KillCountService.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    }
+
+    public void onLoot(LootReceived event) {
+        boolean increment;
+        if (event.getType() == LootRecordType.NPC && "The Whisperer".equalsIgnoreCase(event.getName())) {
+            // Special case: upstream fires LootReceived for the whisperer, but not NpcLootReceived
+            increment = true;
+        } else {
+            increment = event.getType() == LootRecordType.EVENT;
+        }
```
Comment (pajlada) [inline src/main/java/dinkplugin/util/KillCountService.java:73]: code style question:
Could this be a switch instead?

```
boolean increment = false;
switch (event.getType()) {
  case LootRecordType.NPC:
    increment = "The Whisperer".equalsIgnoreCase(event.getName());
    break;
  case LootRecordType.Event:
    increment = true;
    break;
}
```

I don't know if that's very un-java, but that reads easier for me

File: src/main/java/dinkplugin/util/KillCountService.java [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+                killCounts.asMap().merge(boss, kc, Math::max);
+            }, 15, TimeUnit.SECONDS);
+        });
+    }
+
+    @Nullable
+    public Integer getKillCount(LootRecordType type, String sourceName) {
+        if (sourceName == null || (type != LootRecordType.NPC && type != LootRecordType.EVENT)) {
+            return null;
+        }
```
Comment (pajlada) [inline src/main/java/dinkplugin/util/KillCountService.java:102]: Split this up into two short-circuits

**Review by pajlada — APPROVED** [review]
Seems good to me! :+1: I have not tested this

**Review by pajlada — APPROVED** [review]

**Comment by iProdigy** [comment]
Passes preliminary tests, merging now so follow-up PR for #371 can be immediately opened (and comprehensive in-game testing can be done there)

**Pull request merged**